mixer/adapter/kubernetesenv/kubernetesenv.go

```diff
@@ -129,26 +133,108 @@ func (b *builder) Build(ctx context.Context, env adapter.Env) (adapter.Handler,
 		if err != nil {
 			return nil, fmt.Errorf("could not build kubernetes client: %v", err)
 		}
-		controller = newCacheController(clientset, refresh, env)
-		env.ScheduleDaemon(func() { controller.Run(stopChan) })
-		// ensure that any request is only handled after
-		// a sync has occurred
-		env.Logger().Infof("Waiting for kubernetes cache sync...")
-		if success := cache.WaitForCacheSync(stopChan, controller.HasSynced); !success {
-			stopChan <- struct{}{}
-			return nil, errors.New("cache sync failure")
+
+		controller, err = getNewCacheController(b, clientset, env)
+		if err != nil {
+			return nil, fmt.Errorf("could not create new cache controller: %v", err)
 		}
-		env.Logger().Infof("Cache sync successful.")
+
+		controllers = append(controllers, controller)
 		b.controllers[path] = controller
+
+		remote_controllers, err := createRemoteCacheControllers(b, clientset, env)
```

Comment:
If you have to rework this for any reason please add a comment WRT to what you expect createRemoteCacheControllers to return.   It took me a few minutes to realize this might return "nil" for remote_controllers if there are no remotes and essentially add "nil" to the controllers list.   It was a bit obtuse.